fix: Improve data retention wording PUL-2190 (#1820)
codacy/docs: docs/organizations/managing-security-and-risk.md

## Data retention

> Codacy retains all items indefinitely, except in the cases below.

Reviewer: I would put it the other way around, to avoid readers retaining only the first part. 😉
WDYT?

```suggestion
Except for the situations described below, Codacy retains all items indefinitely.
```
